pose-detection/src/posenet/ops/get_offset_vectors_webgpu.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import '@tensorflow/tfjs-backend-webgl';
+import * as tfwebgpu from '@tensorflow/tfjs-backend-webgpu';
+import * as tf from '@tensorflow/tfjs-core';
+
+function getMainHeaderString(index: string): string;
+function getMainHeaderString(...params: string[]): string {
+  let snippet: string;
+  switch (params.length) {
+    case 0:
+      snippet = `fn main() `;
+      break;
+    case 1:
+      snippet = `fn main(${params[0]} : i32)`;
+      break;
+    default:
+      throw Error('Unreachable');
+  }
+  return snippet;
+}
+
+class GetOffsetVectorsProgram implements tfwebgpu.WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  // A is heatmapScores, B is heatMapCoords.
+  variableNames = ['A', 'B'];
+  workgroupSize: [number, number, number];
+  size = true;
+  supportedLastDimension = 2;
+
+  constructor(outputShape: number[]) {
+    // Only 2d tensor whose last dimension is 2 is supported.
+    if (outputShape.length !== 2 ||
+        outputShape[1] !== this.supportedLastDimension) {
+      throw new Error(`GetOffsetVectorsProgram only supports shape of [x, ${
+          this.supportedLastDimension}], but current shape is ${outputShape}`);
+    }
+    const workgroupSizeX = 4;
```

Comment:
Usually, we use `64` as the workgroup size. At least, we should choose a workgroup size larger than the simd length(32).

Reply:
This class is strongly bounded with webgpu internal implementation. Not sure whether we should use the standard webgpu interface to implement it?

Reply:
Changed to 32.